homeassistant/components/workday/__init__.py

```diff
@@ -1 +1,95 @@
 """Sensor to indicate whether the current day is a workday."""
+from __future__ import annotations
+
+from typing import Any
+
+import holidays
+from holidays import HolidayBase
+import voluptuous as vol
+
+from homeassistant.const import CONF_NAME, WEEKDAYS, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv, discovery
+from homeassistant.helpers.typing import ConfigType
+
+DOMAIN = "workday"
+
+ALLOWED_DAYS = WEEKDAYS + ["holiday"]
+
+CONF_COUNTRY = "country"
+CONF_PROVINCE = "province"
+CONF_WORKDAYS = "workdays"
+CONF_EXCLUDES = "excludes"
+CONF_OFFSET = "days_offset"
+CONF_ADD_HOLIDAYS = "add_holidays"
+CONF_REMOVE_HOLIDAYS = "remove_holidays"
+
+# By default, Monday - Friday are workdays
+DEFAULT_WORKDAYS = ["mon", "tue", "wed", "thu", "fri"]
+# By default, public holidays, Saturdays and Sundays are excluded from workdays
+DEFAULT_EXCLUDES = ["sat", "sun", "holiday"]
+DEFAULT_NAME = "Workday Sensor"
+DEFAULT_OFFSET = 0
+
+
+def valid_country(value: Any) -> str:
+    """Validate that the given country is supported."""
+    value = cv.string(value)
+    all_supported_countries = holidays.list_supported_countries()
+
+    try:
+        raw_value = value.encode("utf-8")
+    except UnicodeError as err:
+        raise vol.Invalid(
+            "The country name or the abbreviation must be a valid UTF-8 string."
+        ) from err
+    if not raw_value:
+        raise vol.Invalid("Country name or the abbreviation must not be empty.")
+    if value not in all_supported_countries:
+        raise vol.Invalid("Country is not supported.")
+    return value
+
+
+DATE_VALIDATOR = HolidayBase().__keytransform__
```

Comment:
I'm not sure if it's allowed (and it's definitely a questionable practice) to call into a dunder method of another package, but this allows us to use the validation logic of the holidays package instead of just requiring a string and failing at setup time.